examples/app-actions/src/lib/contentful-context.ts

```diff
@@ -0,0 +1,96 @@
+import { CanonicalRequest, getManagementToken, verifyRequest } from '@contentful/node-apps-toolkit';
+import { IncomingHttpHeaders } from 'http';
+import { StatusCodes } from 'http-status-codes';
+import { Handler } from 'express';
+
+import { sendError } from './utils';
+import { createClient } from 'contentful-management';
+
+const DEFAULT_TTL = 300;
+
+const parseHeaders = (headers: IncomingHttpHeaders = {}): Record<string, string> => {
+  const result: Record<string, string> = {};
+  for (const [header, value] of Object.entries(headers)) {
+    if (Array.isArray(value)) {
+      result[header] = value[0].toString();
+    } else if (value) {
+      result[header] = value.toString();
+    }
+  }
+  return result;
+};
+
+const parseBody = (x: unknown) => (typeof x === 'string' ? x : JSON.stringify(x));
+
+interface MakeContentfulContextMiddlewareOptions {
+  appDefinitionId: string;
+  signingSecret: string;
+  privateKey: string;
+}
+
+export function contentfulContext({
+  signingSecret,
+  appDefinitionId,
+  privateKey,
+}: MakeContentfulContextMiddlewareOptions): Handler {
+  return async (request, response, next) => {
+    const headers = parseHeaders(request.headers);
+
+    const isVerified = verifyRequest(
+      signingSecret,
+      {
+        method: request.method as CanonicalRequest['method'],
+        path: request.url,
+        headers,
+        body: parseBody(request.body),
```

Comment:
Does this actually work when `body` is an object? This assumes that `JSON.parse` returns the same string which is stringified by the Contentful servers.

Reply:
I think the name `parseBody` is misleading here (just because `JSON.parse` is the opposite of what's really happening), but the verification is done comparing strings, so this should be working. 

We could however be a little safer by removing whitespaces

Reply:
Rename it to `stringifyBody`?

Reply:
In my experience with request verification, you always want to create the signature based on the raw body which is not yet parsed in any way. With the `JSON.stringify` you try to recreate the raw body in case it was accidentally already parsed. This is likely to cause issues as afaik parsing and stringifying an object does not guarantee to output the same string. Also, if there is any indentation in the raw body, things will break anyway.

Reply:
Yeah, I agree: request verification heavily relies on the developer having "done their homework" and have the body in a shape that allows for verifying the signature. 

This middleware is by no means exhaustive of all the possible cases, however it runs on NextJS which does the stringification for you anyway. If we think this check makes the code more confusing, we can just get rid of it altogether

Reply:
I'd personally get rid of it. We build a Next.js app, so we should only do whatever is required for Next.js.
This might be different if we have an example built for request verification.

Reply:
Removed it and now passing plain body there as-is.